From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Quentin Kaiser <quentin.kaiser@onekey.com>
Date: Wed, 21 Feb 2024 17:01:24 +0100
Subject: [PATCH] 6246: Fix BSIM listexes command line options handling.

Although support for the '--limit' option is implemented, it's not
possible to use it due to its absence from the set of valid options
(GET_EXECUTABLES_OPTIONS).

This commit fix it by adding LIMIT_OPTION to GET_EXECUTABLES_OPTIONS
set.
---
 .../java/ghidra/features/bsim/query/ingest/BSimLaunchable.java  | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Features/BSim/src/main/java/ghidra/features/bsim/query/ingest/BSimLaunchable.java b/Ghidra/Features/BSim/src/main/java/ghidra/features/bsim/query/ingest/BSimLaunchable.java
index d0d23ae04..a03d5a814 100644
--- a/Ghidra/Features/BSim/src/main/java/ghidra/features/bsim/query/ingest/BSimLaunchable.java
+++ b/Ghidra/Features/BSim/src/main/java/ghidra/features/bsim/query/ingest/BSimLaunchable.java
@@ -157,7 +157,7 @@ public class BSimLaunchable implements GhidraLaunchable {
 	private static final Set<String> LIST_FUNCTIONS_OPTIONS = 
 			Set.of(MD5_OPTION, NAME_OPTION, ARCH_OPTION, COMPILER_OPTION, PRINT_SELF_SIGNIFICANCE_OPTION, CALL_GRAPH_OPTION, PRINT_JUST_EXE_OPTION, MAX_FUNC_OPTION);
 	private static final Set<String> GET_EXECUTABLES_OPTIONS = 
-			Set.of(MD5_OPTION, NAME_OPTION, ARCH_OPTION, COMPILER_OPTION, SORT_COL_OPTION, INCLUDE_LIBS_OPTION);
+			Set.of(MD5_OPTION, NAME_OPTION, ARCH_OPTION, COMPILER_OPTION, SORT_COL_OPTION, LIMIT_OPTION, INCLUDE_LIBS_OPTION);
 	private static final Set<String> GET_EXECUTABLES_COUNT_OPTIONS = 
 			Set.of(MD5_OPTION, NAME_OPTION, ARCH_OPTION, COMPILER_OPTION, INCLUDE_LIBS_OPTION);
 	//@formatter:on
-- 
2.43.0

